crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -152,57 +161,308 @@ impl RawModuleResolutionSettings {
             custom_typeshed,
         } = self;
 
-        let mut paths: Vec<ModuleResolutionPathBuf> = extra_paths
+        let extra_paths: Vec<SearchPathRoot> = extra_paths
             .into_iter()
-            .map(|fs_path| ModuleResolutionPathBuf::extra(fs_path).unwrap())
+            .map(|fs_path| {
+                Arc::new(
+                    ModuleResolutionPathBuf::extra(SystemPath::absolute(
+                        fs_path,
+                        current_directory,
+                    ))
+                    .unwrap(),
+                )
+            })
             .collect();
 
-        paths.push(ModuleResolutionPathBuf::first_party(workspace_root).unwrap());
-
-        paths.push(
-            custom_typeshed.map_or_else(ModuleResolutionPathBuf::vendored_stdlib, |custom| {
-                ModuleResolutionPathBuf::stdlib_from_custom_typeshed_root(&custom).unwrap()
-            }),
+        let workspace_root = Arc::new(
```

Comment:
Yeah, that's a great point

Reply:
I'll defer fixing this to a followup, since calling this `first_party_root` consistently would mean several files untouched by this PR would be altered, and it's a preexisting issue

Reply:
Sounds good

Reply:
I filed https://github.com/astral-sh/ruff/issues/12337 so I don't forget